### PR TITLE
Update FleCSPH package

### PR DESCRIPTION
**Pull request by JulienLoiseau**
- Change to use most recent FleCSI. 
- Add variants

### CODE REVIEW

**Comment by JulienLoiseau** [comment]
@tgamblin This will fix the problem for FleCSI, removing the 1.4 version.

File: var/spack/repos/builtin/packages/flecsph/package.py [hunk 1]
```diff
@@ -15,24 +14,20 @@ class Flecsph(CMakePackage):
...
 
-    homepage = "http://flecsi.lanl.com"
+    homepage = "http://flecsi.org"
     git = "https://github.com/laristra/flecsph.git"
 
+    maintainers("JulienLoiseau")
     version("master", branch="master", submodules=True, preferred=True)
 
-    variant("test", default=True, description="Adding tests")
+    variant("test", default=False, description="Enable unit testing")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/flecsph/package.py]: This shouldn't be a variant (it doesn't result in different binaries being built).

File: var/spack/repos/builtin/packages/flecsph/package.py [hunk 2]
```diff
@@ -41,11 +36,11 @@ def setup_run_environment(self, env):
...
-        options.append("-DLOG_STRIP_LEVEL=1")
-        options.append("-DENABLE_UNIT_TESTS=ON")
-        options.append("-DENABLE_DEBUG_TREE=OFF")
-        # add option to build the tests
+        options = ["-DLOG_STRIP_LEVEL=1"]
+        if "+test" in spec:
+            options.append("-DENABLE_UNIT_TESTS")
+        if "+debug_tree" in spec:
+            options.append("-DENABLE_DEBUG_TREE=ON")
+            options.append("-DENABLE_DEBUG=ON")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/flecsph/package.py]:
```suggestion
        options = [
            self.define("LOG_STRIP_LEVEL", True),
            self.define(("ENABLE_UNIT_TESTS", self.run_tests),
            self.define_from_variant("ENABLE_DEBUG_TREE", "debug_tree"),
            self.define_from_variant("ENABLE_DEBUG", "debug_tree"),
        ]
```

File: var/spack/repos/builtin/packages/flecsph/package.py (hunk 1 above)
Comment (alalazo) [inline var/spack/repos/builtin/packages/flecsph/package.py]:
```suggestion
```

**Review by alalazo — CHANGES_REQUESTED** [review]
A few suggestions to improve the recipe.

File: var/spack/repos/builtin/packages/flecsph/package.py (hunk 1 above)
Comment (JulienLoiseau) [reply var/spack/repos/builtin/packages/flecsph/package.py]: I removed the variant.

File: var/spack/repos/builtin/packages/flecsph/package.py (hunk 2 above)
Comment (JulienLoiseau) [reply var/spack/repos/builtin/packages/flecsph/package.py]: I made the proposed changes.

File: var/spack/repos/builtin/packages/flecsph/package.py (hunk 1 above)
Comment (JulienLoiseau) [reply var/spack/repos/builtin/packages/flecsph/package.py]: I made the proposed changes.

File: var/spack/repos/builtin/packages/flecsph/package.py [hunk 3]
```diff
@@ -15,37 +14,32 @@ class Flecsph(CMakePackage):
...
-    depends_on("boost@1.70.0: cxxstd=17 +program_options")
-
+    depends_on("boost@1.70.0: +atomic +filesystem +regex +system")
     depends_on("mpi")
     depends_on("hdf5+hl@1.8:")
-    depends_on("flecsi@1.4.2 +external_cinch backend=mpi")
+    depends_on("flecsi@2.2 +flog backend=mpi")
     depends_on("gsl")
-    depends_on("googletest", when="+test")
+    depends_on("googletest")
```
Comment (rbberger) [inline var/spack/repos/builtin/packages/flecsph/package.py]:
```suggestion
    depends_on("googletest", type="test")
```

**Comment by rbberger** [comment]
@alalazo is there something still missing?

**Review by alalazo — APPROVED** [review]

**Pull request merged**